src/widgets/scopes/videowaveformscopewidget.cpp

```diff
@@ -82,17 +87,65 @@ void VideoWaveformScopeWidget::paintEvent(QPaintEvent*)
     if (!isVisible())
         return;
 
+    // Create the painter
     QPainter p(this);
     p.setRenderHint(QPainter::Antialiasing, true);
+    QFont font = QWidget::font();
+    int fontSize = font.pointSize() - (font.pointSize() > 10? 2 : (font.pointSize() > 8? 1 : 0));
+    font.setPointSize(fontSize);
+    QFontMetrics fm(font);
+    p.setPen(palette().text().color().rgb());
+    p.setFont(font);
+
+    // Fill the background
     p.fillRect(0, 0, width(), height(), QBrush(Qt::black, Qt::SolidPattern));
+
+    // draw the waveform data
     m_mutex.lock();
     if(!m_displayImg.isNull()) {
         p.drawImage(rect(), m_displayImg, m_displayImg.rect());
     }
     m_mutex.unlock();
+
+    // Add IRE lines
+    int textpad = 3;
+    // 100
+    qreal ire100y = height() - (height() * IRE100 / 255);
+    p.drawLine(QPointF(0, ire100y), QPointF(width(), ire100y));
+    p.drawText(textpad, ire100y - textpad, tr("100"));
+    // 0
+    qreal ire0y = height() - (height() * IRE0 / 255);
+    p.drawLine(QPointF(0, ire0y), QPointF(width(), ire0y));
+    QRect textRect = fm.tightBoundingRect(tr("0"));
+    p.drawText(textpad, ire0y + textRect.height() + textpad, tr("0"));
+
     p.end();
 }
 
+void VideoWaveformScopeWidget::mouseMoveEvent(QMouseEvent *event)
+{
+    QString text;
+    qreal ire100y = height() - (height() * IRE100 / 255);
+    qreal ire0y = height() - (height() * IRE0 / 255);
+    qreal ireStep = (ire0y - ire100y) / 100.0;
+    int ire = (ire0y - event->pos().y()) / ireStep;
+
+    m_mutex.lock();
+    int frameWidth = m_displayImg.width();
+    m_mutex.unlock();
+
+    if(frameWidth != 0)
+    {
+        int pixel = frameWidth * event->pos().x() / width();
+        text =  QString("Pixel: %1\nIRE: %2").arg(QString::number(pixel), QString::number(ire));
```

Comment:
Should this string be translatable?

Reply:
https://github.com/mltframework/shotcut/pull/669